services-api/src/main/java/io/scalecube/services/Reflect.java

```diff
@@ -270,7 +297,14 @@ public static void validateMethodOrThrow(Method method) {
       throw new UnsupportedOperationException("Service method return type can be Publisher only");
     }
     if (method.getParameterCount() > 1) {
-      throw new UnsupportedOperationException("Service method can accept 0 or 1 parameters only");
+      int paramsCount = 0;
+
+      for (Parameter parameter : method.getParameters()) {
```

Comment:
why don't you do an index loop?

Reply:
This part of the code doesn't exist anymore.